Missing Gopkg.lock causes crash

In the case that `Gopkg.toml` exists but `Gopkg.lock` does, not, nancy crashes with a nil pointer dereference:

```
Nancy version: development
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x48 pc=0x935557]

goroutine 1 [running]:
github.com/sonatype-nexus-community/nancy/packages.ExtractPurlsUsingDep(0xc00002a124, 0x2b, 0xc00002b170, 0x2b, 0xc00002a139, 0x16, 0xc0001749a0, 0x0, 0x0, 0xc00002a139, ...)
        /home/jonhall/go/src/github.com/sonatype-nexus-community/nancy/packages/dep.go:23 +0x37
main.doCheckExistenceAndParse()
        /home/jonhall/go/src/github.com/sonatype-nexus-community/nancy/main.go:102 +0x214
main.main()
        /home/jonhall/go/src/github.com/sonatype-nexus-community/nancy/main.go:63 +0x104
```

To reproduce, create an empty directory containing a `Gopkg.toml` file, then attempt to run nancy.  Example:

```shell
$ mkdir asdf
$ cd asdf
asdf$ touch Gopkg.toml
asdf$ nancy Gopkg.lock
```

--- a/main.go
+++ b/main.go
@@ -15,6 +15,7 @@ package main
 
 import (
 	"bufio"
+	"errors"
 	"flag"
 	"fmt"
 	"os"
@@ -96,7 +97,10 @@ func doCheckExistenceAndParse() {
 		}
 		project, err := ctx.LoadProject()
 		if err != nil {
-			customerrors.Check(err, fmt.Sprint("could not read lock at path "+config.Path))
+			customerrors.Check(err, fmt.Sprint("could not read lock at path " + config.Path))
+		}
+		if project.Lock == nil {
+			customerrors.Check(errors.New("dep failed to parse lock file and returned nil"), "nancy could not continue due to dep failure")
 		}
 
 		purls, invalidPurls := packages.ExtractPurlsUsingDep(*project)

